### PR TITLE
Send files from server in compressed format

### PR DESCRIPTION
**Pull request by vivonk**
## Description
In before, files were getting sent in an uncompressed format which uses so much internet. Now files get send after it gets compressed and user don't have to wait for the time of compression. As soon compression will get done, client side will make request to get the compressed one file

## Related Issue
#455 

## Motivation and Context

Files are around here and there but the best way to manage and transfer a file is considered where data consumance would be low.

## How Has This Been Tested?

Tested with multiple browsers, used multiple systems at same to check whether files are not getting compressed twice for the same

## Screenshots (In case of UI changes):
UI have got just a single Toast of compression in progress, else things are in the hole 😊 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

## TODO
- [ ] Update main documentation regarding temporary folders
- [ ] Implement some Admin services to delete temporary files as we can't have any logic for deleting files
- [ ] Add changes to UI for multiple files download and actions on multiple files

### CODE REVIEW

File: components/core/DownloadManager.py [hunk 1]
```diff
@@ -7,367 +7,437 @@
...
+		try:
+			cursor.execute(sql, (download.link, download.userName, int(time.time()), download_name))
+			db.commit()
+		except MySQLdb.Error as e:
+			db.rollback()
+			# Shows error thrown up by database
+			print(e)
+			return e[1]
+		return "success"
+	return "db connection error"
```
Comment (vivonk) [inline components/core/DownloadManager.py]: It shows things right on editing the file with indentation 4, here is show 8 where my original files consider it as 4
@agentmilindu Please suggest something

**Comment by vivonk** [comment]
@agentmilindu please review this one. It has a good structure for compression work (actual background process of compress) and asynchronous file request handling is also added (for same or multiple files)

**Comment by vivonk** [comment]
I am making a new issue about how to delete the file. I was considering the administration role here because only admin could manage files (as standards could be defined)

**Comment by vivonk** [comment]
@agentmilindu  should we keep it open or merge it before we refactor the code :)

**Comment by agentmilindu** [comment]
@vivonk Let's merge it first and then refactor.

**Comment by vivonk** [comment]
@agentmilindu I have refactored till #454 but I will add these modules in relevant PR. Yes, we can merge

File: components/core/file_sender/sender.py [hunk 2]
```diff
@@ -1,28 +1,89 @@
...
+			logging.error(" get file API (/api/file) got wrong arguments, thrown an error :: %s" % e)
+			return Response("error", status=200)
+		
+		try:
+			return send_file(filename_or_fp=download_path, attachment_filename=file_name,
+							 as_attachment=True)
+		except Exception as e:
+			logging.error("File sending has a exception. When sending file, we got :: %s" % e)
+			return Response("File you are trying to access is not available to us. Please ask admin to check the server"
+							, status=200)
```
[the remark this answers is not in the document]
Comment (vivonk) [reply components/core/file_sender/sender.py]: Yes, as file is not found so internal tools are having somewhere problem. But same thing can happen like someone manually deleted file so for that instance, we have to show file not found, 404. what you say? @JaDogg

File: components/core/file_sender/sender.py [hunk 3]
```diff
@@ -1,28 +1,89 @@
...
+	token = token_validator(request.args['token'])
+	if token is not None:
+		try:
+			gid = request.args.get('gid')
+			logging.info('User asked for file, having GID (ugid) ' + gid)
+			file_name = file_name_formatter(gid) + '.zip'
+			download_path = get_zip_path(gid)
+		except Exception as e:
+			logging.error(" get file API (/api/file) got wrong arguments, thrown an error :: %s" % e)
+			return Response("error", status=200)
```
[the remark this answers is not in the document]
Comment (vivonk) [reply components/core/file_sender/sender.py]: Yes, I will correct it. Thanks @JaDogg

File: components/core/file_sender/sender.py [hunk 4]
```diff
@@ -1,28 +1,89 @@
...
+		conf_file.close()
+	return json.loads(conf_str)
+
+
+def get_zip_path(file_name):
+	conf_json = dl_config_reader()
+	tmp_dir = conf_json['tmp_folder']
+	if not os.path.exists(tmp_dir):  # check tmp directory exists or not
+		os.makedirs(tmp_dir)
+	zip_path = tmp_dir + '/' + file_name_formatter(file_name) + '.zip'
```
[the remark this answers is not in the document]
Comment (vivonk) [reply components/core/file_sender/sender.py]: ok let me give it a try with `os.path.join`

**Comment by JaDogg** [comment]
Good stuff. 👍

**Comment by vivonk** [comment]
Thanks

File: components/core/file_sender/sender.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (JaDogg) [reply components/core/file_sender/sender.py]: yeah 👍

File: components/core/file_sender/compression_handler.py [hunk 5]
```diff
@@ -42,7 +42,7 @@ def get_zip_path(file_name):
 	tmp_dir = conf_json['tmp_folder']
 	if not os.path.exists(tmp_dir):  # check tmp directory exists or not
 		os.makedirs(tmp_dir)
-	zip_path = tmp_dir + '/' + file_name_formatter(file_name) + '.zip'
+	zip_path = os.path.join(tmp_dir, file_name_formatter(file_name) + '.zip')
```
Comment (JaDogg) [inline components/core/file_sender/compression_handler.py]: Nice 👍

**Comment by vivonk** [comment]
@agentmilindu can we merge this :smile: >

**Pull request merged**